tests/end_to_end/test_server.py

```diff
@@ -112,3 +107,52 @@ def getServerErrLines(self):
         self.serverErrFile.seek(0)
         serverErrLines = self.serverErrFile.readlines()
         return serverErrLines
+
+    def getServerCmdLine(self):
+        serverCmdLine = """
+            python server_dev.py
+            --dont-use-reloader
+            --config TestConfig
+            --port {} """.format(self.port)
+        return serverCmdLine
+
+
+class ServerTestConfigFile(ServerTest):
+    """
+    Launch a server test, but with a custom configuration file
+    """
+    def run(self, *args, **kwargs):
+        self.config = None
+        self.configFile = None
+        self.configFilePath = None
+        super(ServerTestConfigFile, self).run(*args, **kwargs)
+
+    def setConfig(self):
+        raise NotImplementedError("ServerTestConfigFile must be subclassed")
+
+    def getServerCmdLine(self):
+        self.setConfig()
+        self.configFile = tempfile.NamedTemporaryFile()
+        self.configFile.write(self.config)
+        self.configFile.flush()
+        self.configFilePath = self.configFile.name
+        serverCmdLine = """
+            python server_dev.py
+            --dont-use-reloader
+            --config TestConfig
+            --config-file {}
+            --port {} """.format(self.configFilePath, self.port)
+        return serverCmdLine
+
+
+class ServerTestConfigFileDeterministic(ServerTestConfigFile):
+    """
+    Launch a server test with a custom configuration file
+    that ensures there is no pseudorandom behavior.
+    """
+    def setConfig(self):
+        self.config = """
+SIMULATED_BACKEND_NUM_VARIANT_SETS = 1
```

Comment:
Then you override `ServerTestConfigFile` and implement `setConfig` in your subclass with 5 variantsets and whatever else you want in the config.